src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(object: T[], key: keyof T): GroupsMap<T> {
+  const finalObject: GroupsMap<T> = {};
+
+  object.forEach((el) => {
+    const keyValue = el[key];
```

Comment:
The variable `keyValue` is inferred to be of type `T[keyof T]`, which might not be a string or number. You should ensure that the key used to group the elements is a property with a string or number value to avoid runtime errors when converting it with `String(keyValue)`.